test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -408,18 +419,58 @@ void UcxContext::destroy_worker()
     ucp_worker_destroy(_worker);
 }
 
+ucs_status_t UcxContext::am_recv_callback(void *arg, const void *header,
+                                          size_t header_length,
+                                          void *data, size_t length,
+                                          const ucp_am_recv_param_t *param)
+{
+    UcxContext *self = reinterpret_cast<UcxContext*>(arg);
+
+    assert(param->recv_attr & UCP_AM_RECV_ATTR_FIELD_REPLY_EP);
+
+    uint64_t id = reinterpret_cast<uint64_t>(param->reply_ep);
+    conn_map_t::iterator iter = self->_conns.find(id);
+    if (iter == self->_conns.end()) {
```

Comment:
it could be assertion, since we use conn request API on server side. server cannot receive AM before server has `accepted` conn_request and the same for client.

Reply:
yes, but what if connection got deleted before we polled this data?

Reply:
in this case the EP has to be destroyed as well and all the data should be dropped

Reply:
how would you drop rx message?

Reply:
see UCP stream impl

Reply:
can you please be more specific? AFAIU, in AM this is the way to drop the data - you just releases it by returning OK to UCT

Reply:
if the EP is in closing state - no need to invoke callback since it is not operable

Reply:
where is it handled?